hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieKeyBasedFileGroupRecordBuffer.java

```diff
@@ -127,10 +124,12 @@ public boolean hasNext() throws IOException {
 
       String recordKey = readerContext.getRecordKey(baseRecord, baseFileSchema);
       Pair<Option<T>, Map<String, Object>> logRecordInfo = records.remove(recordKey);
+      Map<String, Object> metadata = readerContext.generateMetadataForRecord(
+          baseRecord, baseFileSchema, false);
```

Comment:
Caution for the performace regression for per-record metadata construction.

Reply:
We'll do benchmarking on this.  Likely it's ok given we're not doing much work inside the method and the existing also extract some metadata for each record.